src/libs/swapAndBridge/swapAndBridge.ts

```diff
@@ -130,30 +174,65 @@ const buildSwapAndBridgeUserRequests = (
   const requests: SignUserRequest[] = []
   if (userTx.approvalData) {
     const erc20Interface = new Interface(ERC20.abi)
-    requests.push({
-      id: `${userTx.activeRouteId}-approval`,
-      action: {
-        kind: 'calls' as const,
-        calls: [
-          {
-            to: userTx.approvalData.approvalTokenAddress,
-            value: BigInt('0'),
-            data: erc20Interface.encodeFunctionData('approve', [
-              userTx.approvalData.allowanceTarget,
-              BigInt(userTx.approvalData.minimumApprovalAmount)
-            ]),
-            fromUserRequestId: `${userTx.activeRouteId}-approval`
-          } as Call
-        ]
-      },
-      meta: {
-        isSignAction: true,
-        networkId,
-        accountAddr: account.addr,
-        activeRouteId: userTx.activeRouteId,
-        isApproval: true
-      }
-    } as SignUserRequest)
+    let shouldApprove = true
+    try {
+      const erc20Contract = new Contract(
+        userTx.approvalData.approvalTokenAddress,
+        ERC20.abi,
+        provider
+      )
+      const allowance = await erc20Contract.allowance(
+        userTx.approvalData.owner,
+        userTx.approvalData.allowanceTarget
+      )
+      // check if an approval already exists
+      if (BigInt(allowance) === MaxUint256) shouldApprove = false
```

Comment:
This is good, but it can be better. Let's decide should we pursue such a case.  
Currently, if the amount is infinite, it won't ask for an approval, which is great. But what if I have approved 2000 DAI prior to coming to our swap and decide to swap&bridge 100? It will ask me for an infinite approval.  
We can change this to: 
```
if (BigInt(allowance) >== userTx.approvalData.allowanceTarget) shouldApprove = false
```
This won't force an approval until I put an amount bigger than the allowance